pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperCache.java

```diff
@@ -74,11 +79,13 @@
     protected final Cache<String, Set<String>> childrenCache;
     protected final Cache<String, Boolean> existsCache;
     protected final OrderedSafeExecutor executor;
+    protected final ScheduledExecutorService scheduledExecutor;
 
     protected AtomicReference<ZooKeeper> zkSession = new AtomicReference<ZooKeeper>(null);
 
-    public ZooKeeperCache(ZooKeeper zkSession, OrderedSafeExecutor executor) {
+    public ZooKeeperCache(ZooKeeper zkSession, OrderedSafeExecutor executor, ScheduledExecutorService scheduledExecutor) {
```

Comment:
> So, should we introduce a flag or we should create a thread-pool for each ZKCache instance and with what size?

If there is more than 1 instance.. then yes :) a flag would be great. Something like `shouldShutdownExecutor`..